app/models/membership.rb

```diff
@@ -58,7 +60,15 @@ def user_name_or_email
     return user_name ? user_name : user_email
   end
 
+  def discussion_readers
```

Comment:
This should definitely be a has_many relationship, although I haven't quite figured out how to get the right one.
It's something like

```
  has_many :discussion_readers, -> { group: self.group) }, through: :user
```